lib/rack/cache/storage.rb

```diff
@@ -38,6 +38,9 @@ def create_store(type, uri)
         else
           fail "Unknown storage provider: #{uri.to_s}"
         end
+      elsif uri.nil?
+        # hack to support setting entitystore to nil, which doesn't persist request bodies
+        type.const_get(uri.inspect.upcase).resolve(uri)
       else
         # hack in support for passing a Dalli::Client or Memcached object
         # as the storage URI.
```

Comment:
The way backends are instanced in ```Storage#create_store``` is very convoluted, in my opinion. Changing it may make sense, but I think it's outside the scope of this PR. Maybe in another PR.

Changing the way options are passed from clients, so that class instances are accepted instead of uri strings, is in my opinion a bad idea. First, it would break any code that uses this gem if backwards compatibility is dropped; if it is maintained, backend instancing would have to be twice as complicated as it is now. Second, it would expose too much of the gem's internals to client code: why should clients need to be aware that the class that implements file backend is ```Rack::Cache::EntityStore::Disk```?